@xen-orchestra/fs/src/abstract.js

```diff
@@ -388,6 +388,11 @@ export default class RemoteHandlerAbstract {
           info('will update metadata of this remote')
           return this.#createMetadata()
         } else {
+          // to add an new encrypted fs remote, the remote directory must be empty, otherwise metadata.json is not created
+          if (error.code === 'ENOENT' && error.path.includes('metadata.json')) {
+            throw new Error('Remote directory must be empty.')
+          }
```

Comment:
If I understand your PR description correctly, there was already a check and an error, right?

Instead of adding another check and complexifying the code, isn't it possible to catch the previous one at the right place and throw a clearer error instead?

Note: use the `clause` option of `Error` to keep the original error around.

Reply:
There was no check, and the user received a stack trace as an error :
https://xcp-ng.org/forum/topic/9830/nfs-remote-encryption-problem?_=1730189715182

The error is the result of the #canWriteMetadata() being false in the if just above.

Reply:
I'll let @fbeauchamp decide what's best in this case, he can resolve my comment :slightly_smiling_face:

Reply:
I think the check metadata code is overly complex now, but the refactor can be done after the release, thus giving a little more time to test. This commit addresses the core issue